README.md

```diff
@@ -1,2 +1,270 @@
+|osVersion|optional string|e.g. "17.5.1"|The version of the OS. Depending on the OS, they may have different formats to be specified separately.|
+|appVersion|optional string|e.g. "1.0.1"|The version of the app as it is specified on the App/Play Store.|
+|appBuild|optional string|e.g. "56"|The build version of the app as defined by the CI pipeline releasing the app.|
+|language|optional string|e.g. "en"|Following IETF BCP-47 / [FHIR ValueSet languages](https://hl7.org/fhir/R4B/valueset-languages.html)|
+
+Push notifications over APNS / FCM only contain text in a single language. For this, the device's `language` property shall be prioritized, falling back on the user's `language` property and using US-English if both are not present.
+
+#### Adding a new device or modifying information of an existing one
+
+A device can only be identified by its notification token. When updating a device's information, we therefore check if the notification token already exists in Firestore. If it already exists, we update all other existing fields' values - otherwise, we create a new device.
+
+A device may receive a different notification token at any time though. Therefore, we might create a new device, even though that device already exists in the table with a different (now inactive) notification token. Therefore, every time we send out a new notification and receive the information that a token is no longer active, we need to remove the device from this table.
+
+### /users/$userId$/messages/$messageId$
+
+|Property|Type|Values|Comments|
+|-|-|-|-|
+|dueDate|optional DateTime|-|The due date of the message to be shown in-app.|
+|triggerDate|optional DateTime|-|A date to trigger a notification on the device. If not set, then there is no notification to trigger. The notification should be triggered by the server. TBD: Are there cases where a message would trigger multiple times or event- rather than date-driven?|
+|completionDate|optional DateTime|-|Specifies when a message has been completed. TBD: Messages containing a completionDate may either be hidden on user's devices or be shown crossed out.|
+|type|optional string|e.g. "questionnaireReminder"|Some messages are sent out on a regular basis, where only the most recent message is really relevant for the patient (e.g. a reminder for a questionnaire). With this property, we can easily find existing messages of the same type and replace them with a new one, if necessary.|
+|title|LocalizedText|e.g. "Watch Welcome Video in Education Page."|May be localized.|
+|description|optional LocalizedText|e.g. "The video shows how you will be able to use this app."|May be localized.|
+|action|optional string|e.g. "engage-hf:/videoSections/engage-hf/videos/welcome"|See "Available message actions".|
+
+#### Available message actions
+
+TODO: This subsection shall describe the format of actions to be performed by the clients on tap of a message.
```

Comment:
Would be great if we can specify this based on the discussions we had in the meeting today & UI document.

Reply:
I changed the action section to be a "Message types" section containing both formats for the actions and the available types.